test/framework/management/kind/mgmt.go

```diff
@@ -112,19 +107,56 @@ func (c *Cluster) GetName() string {
 
 // LoadImage will put a local image onto the kind node
 func (c *Cluster) LoadImage(ctx context.Context, image string) error {
-	loadCmd := exec.NewCommand(
-		exec.WithCommand("kind"),
-		exec.WithArgs("load", "docker-image", image, "--name", c.Name),
+	provider := cluster.NewProvider(
+		cluster.ProviderWithLogger(cmd.NewLogger()),
 	)
-	stdout, stderr, err := loadCmd.Run(ctx)
+
+	// Save the image into a tar
+	dir, err := fs.TempDir("", "image-tar")
+	if err != nil {
+		return errors.Wrap(err, "failed to create tempdir")
+	}
+	defer os.RemoveAll(dir)
+	imageTarPath := filepath.Join(dir, "image.tar")
+
+	err = save(ctx, image, imageTarPath)
+	if err != nil {
+		return err
+	}
+
+	nodeList, err := provider.ListInternalNodes(c.Name)
 	if err != nil {
-		fmt.Println(string(stdout))
-		fmt.Println(string(stderr))
 		return err
 	}
+
+	// Load the image on the selected nodes
+	for _, node := range nodeList {
+		if err := loadImage(imageTarPath, node); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+// save saves image to dest, as in `docker save`
+func save(ctx context.Context, image, dest string) error {
```

Comment:
Perhaps `saveImage` to be consistent with `loadImage`?
```suggestion
func saveImage(ctx context.Context, image, dest string) error {
```

Reply:
I don't mind save or save image as it's an internal function with a comment, but i do agree saveImage is more consistent.

Ah, I noticed this was lifted from kind. 

https://github.com/kubernetes-sigs/kind/blob/master/pkg/cmd/kind/load/docker-image/docker-image.go#L158-L170

Reply:
should we keep it `save` as it is directly copied from kind and maybe add a comment? but saveImage is more consistent

Reply:
it's fine how it is imo